### PR TITLE
[FIX] survey: compute matrix answers display_name too

### PR DESCRIPTION
**Pull request by flch-odoo**
We here fix the `SurveyQuestionAnswer._compute_display_name` method introduced in 55fa52be.

`survey.question.answers` used as matrix rows and columns require different treatment as they are not used in triggers but are both shown on the `survey.user.input.line` views, where the display shouldn't change (nor cause a crash).

It also doesn't make much sense to create answers outside the context of a question, so we remove the button that already wasn't shown on the tree view.

As users may not fully upgrade their views though, we added a fallback question title in `compute_display_name`too.

Task-3495142

### CODE REVIEW

File: addons/survey/models/survey_question.py [hunk 1]
```diff
@@ -681,17 +681,20 @@ class SurveyQuestionAnswer(models.Model):
...
         This implementation makes sure we have at least 30 characters for the question title,
         then we elide it, leaving the rest of the space for the answer.
         """
         for answer in self:
-            # _origin (or fallback title) is (likely temporarily) needed to support survey snapshot
-            # during onchange for a deleted answer used as trigger in another question.
-            title = answer._origin.question_id.title
+            # _origin is (likely temporarily) needed to support survey snapshot during onchange
+            # for a deleted answer used as trigger in another question.
+            if answer._origin.matrix_question_id or answer._origin.question_id.question_type == 'matrix':
```
Comment (tde-banana-odoo) [inline addons/survey/models/survey_question.py:694]: But why using origin ? It is a compute, it should be based on question_id / matrix_question_id ...

Comment (flch-odoo) [reply addons/survey/models/survey_question.py:694]: We broke something with the normal flow of `onchange` in our survey form view (and questions modal) where we automatically save when we open/close the record (we're pushing the limits here; which we discussed with the JS and python framework). We didn't get to the absolute bottom of it and merged this like this a few weeks ago because it didn't seem worth it when comparing to the 'quick hack'.

Basically, if I remember correctly, it seems like an `onchange` is triggered but it doesn't provide all the data necessary for the compute (because the record hasn't changed?) and so, while `_origin` is correct and contains the right reference(s), `answer.question_id` is `False` during the onchange's snapshot creation, and this causes crashes during our integration test (`survey_tour_test_survey_form_triggers`).
tldr; If we remove `_origin`, we do not get the `question_id` that does exist but `False`. I just confirmed that this is still the case (with all the `api.depends` specified here).

Comment (tde-banana-odoo) [reply addons/survey/models/survey_question.py:694]: It is an issue, we cannot have code that relies on _origin (which is an orm internal trick). If we are pushing the fw too far, maybe we should come back to standard fw. There is no reason a compute should be that hackish ...

Comment (flch-odoo) [reply addons/survey/models/survey_question.py:694]: OK!
I'll discuss this with RCO then, as it is related to the recent refactoring of JS relational model and `onchange2`. 
We didn't think it was worth pursing further (after already several hours) because there are already many instances of using `_origin` in computes and we have good test coverage in case something changes.

**Comment by flch-odoo** [comment]
@ryv-odoo , FYI, the fix we talked about; which currently awaits the completion of your own fix :)

**Comment by flch-odoo** [comment]
Hello @tde-banana-odoo,
I've received the "official" response from the framework that our fallback is sufficient to circumvent the error. We now also know how come there is an error to begin with, and that it is not trivial to fix.

I can explain it in detail on Wednesday, but the conclusion is: 
The case where `question_id` is `False` and where the fallback will be used, is when we're 1. creating a new answer without a question (which we're disabling too, but still, it's correct) or 2. When we are deleting an answer but because it is used in another related field, it's display name still gets computed during an onchange.

So, as you can see, we can successfully remove `_origin` from all places in this method and have our existing and new unit and integration test pass, so it's good to ship :rocket:

**Comment by tde-banana-odoo** [comment]
@robodoo r+

@flch-odoo Thanks for investigating :) would gladly have the complete explanation (but I already send the fix because it is Friday).
